machine-runner/tests/esm/runner.test.ts

```diff
@@ -588,6 +588,40 @@ describe('machine as async generator', () => {
     expect(toggleCount).toBe(2)
   })
 
+  describe('peek/next state store nullification from command persistence', () => {
+    it('should be true that peek/next state store is nullified on command persistence', async () => {
+      const r1 = new Runner(On, { toggleCount: 0 })
+      const { machine } = r1
+      r1.feed([], true) // caughtUp event
+
+      const state = (await machine.peekNext()).value
+      expect(state).toBeTruthy() // should resolve immediately
+      if (!state) throw new Unreachable()
+
+      await r1.caughtUpDelay.toggle(true) // start delay
+      // this command call locks, nullifying state store previously populated by caughtUp event
+      const whenOffCommands = state.as(On)?.commands()
+      if (!whenOffCommands) throw new Unreachable()
+      await whenOffCommands.toggle()
+      await r1.eventRoundtrip.waitAllDone() // make sure all event roundtrips are done
+
+      let nextOrPeekTriggered = false
+      const peek = machine.peekNext()
+      const next = machine.next()
+      Promise.race([peek, next]).finally(() => (nextOrPeekTriggered = true))
```

Comment:
same as we discussed yesterday, which is probably in many other tests as well: after this line we need
```typescript
await Promise.resolve()
```

Reply:
Ah yes. `await r1.eventRoundtrip.waitAllDone()` was below this line; I moved it up, forgetting that it is important down there too.